packages/salesforcedx-vscode-core/src/commands/commands.ts

```diff
@@ -331,12 +331,29 @@ export abstract class SfdxCommandletExecutor<T>
       cwd: getRootWorkspacePath()
     }).execute(cancellationToken);
 
-    execution.processExitSubject.subscribe(() => {
-      this.logMetric(execution.command.logName, startTime);
+    let output = '';
+    execution.stdoutSubject.subscribe(realData => {
+      output += realData.toString();
+    });
+
+    execution.processExitSubject.subscribe(async exitCode => {
+      this.logMetric(
+        execution.command.logName,
+        startTime,
+        await this.getTelemetryData(exitCode === 0, response, output)
```

Comment:
This change will affect all commands that are not implementing their own `execute` method. Is the data for those commands going to look the same as it currently does ?

Reply:
Correct, it will look the same. This is because those commands will use the default getTelemetryData which returns undefined for additional data.

Reply:
I think we need to remove the `await` in `this.getTelemetryData` and `async` in line 340 since we're not calling any promises.

Reply:
This has `await` in case an overridden version requires some asynchronous operation. Awaiting something that's not a promise will just convert the value to a resolved promise.

Reply:
I prefer having this as a sync operation for now and update it to be an async one whenever we have a need for it.